harness/determined/common/api/authentication.py

```diff
@@ -212,24 +159,24 @@ def logout(
     master_address = master_address or util.get_default_master_address()
     token_store = TokenStore(master_address)
 
-    session_user, _ = default_load_user_password(requested_user, None, token_store)
+    user, _ = default_load_user_password(requested_user, None, token_store)
     # Don't log out of DEFAULT_DETERMINED_USER when it's not specified and not the active user.
 
-    if session_user is None:
+    if user is None:
         return
 
-    if session_user == token_store.get_active_user():
+    if user == token_store.get_active_user():
         token_store.clear_active()
 
-    session_token = token_store.get_token(session_user)
+    token = token_store.get_token(user)
 
-    if session_token is None:
+    if token is None:
         return
 
-    token_store.drop_user(session_user)
+    token_store.drop_user(user)
 
-    auth = LogoutAuthentication(session_user, session_token)
-    sess = api.Session(user=session_user, master=master_address, auth=auth, cert=cert)
+    utp = UsernameTokenPair(user, token)
```

Comment:
Per your earlier comment about UTP being a stepping stone, I like what it's done here in that capacity.